airflow/config_templates/default_airflow.cfg

```diff
@@ -125,11 +125,7 @@ sql_alchemy_pool_recycle = 1800
 # Check connection at the start of each connection pool checkout.
 # Typically, this is a simple statement like “SELECT 1”.
 # More information here: https://docs.sqlalchemy.org/en/13/core/pooling.html#disconnect-handling-pessimistic
-sql_alchemy_pool_pre_ping = False
-
-# How many seconds to retry re-establishing a DB connection after
-# disconnects. Setting this to 0 disables retries.
-sql_alchemy_reconnect_timeout = 300
```

Comment:
What happens now with disconnects/errors?

Reply:
If connection from the pool is not healthy, it's removed from the pool and new one is created (with three attempts). Other connections are invalidated from the pool as well. 

https://github.com/sqlalchemy/sqlalchemy/commit/f881dae8179b94f72ab0dc85d8f62be8c9ce2fe0#diff-31816cdb15e64b0af1b862f51abe1226R920 - the test visualizes this. 

This commit from SQLAlchemy is quite a good explanation.

Reply:
pool_pre_ping argument should take care of it and the code for re connect is within sql alchemy library. We don't have to handle explicitly with new sql alchemy version